subsys/net/lib/http/http_server_http1.c

```diff
@@ -313,12 +313,86 @@ static int on_header_field(struct http_parser *parser, const char *at,
 	struct http_client_ctx *ctx = CONTAINER_OF(parser,
 						   struct http_client_ctx,
 						   parser);
+	size_t offset = strlen(ctx->header_buffer);
```

Comment:
Should never occur, but I always tend to prefer the `n` variant when using fixed sized buffers. (If for some obscure reason the trailing `\0` would have been removed).
```suggestion
	size_t offset = strnlen(ctx->header_buffer, sizeof(ctx->header_buffer));
```

Reply:
The buffer should be always null terminated, and even if there null is missing, we have a check in next line to verify that the offset is not overflowing. In that case we also get a printout that something is wrong which we would not get with `strnlen`. I would suggest to leave this as is.

Reply:
The focus is on "should be always null terminated" even reading past the buffer end (which `strlen` will do) can lead to undefined behavior. For example with some `MPU` configurations.

I'm not saying there is an error here, it's just defensive programming.
Oversights might happen if someone were to add code between the check and the handling.

Reply:
There's a reason why modern languages keep length information tied to arrays/slices.

Reply:
Ok, if you feel we need this check, then it can be added of course.